src/reporter/screenshotReporter.js

```diff
@@ -56,17 +58,28 @@ JasmineScreenshotReporter.prototype._registerOnAction = function (options) {
   // screenshot is taken between sync and interaction
   this._registerOnSync();
 
+  var applyAction_ = protractorModule.parent.exports.ElementArrayFinder.prototype.applyAction_;
+  protractorModule.parent.exports.ElementArrayFinder.prototype.applyAction_ = function (cb) {
+    var sLocator = this.locator().toString();
+    var newCb = function (webElem) {
+      return webElem.getAttribute('id').then(function (elementId) {
```

Comment:
this adds a second call to the browser to get the element ID ?. Can we avoid it somehow ?